lldb/packages/Python/lldbsuite/test/tools/lldb-dap/dap_server.py

```diff
@@ -107,6 +107,10 @@ def read_packet_thread(vs_comm, log_file):
             # termination of lldb-dap and stop waiting for new packets.
             done = not vs_comm.handle_recv_packet(packet)
     finally:
+        # Wait for the process to fully exit before dumping the log file to
+        # ensure we have the entire log contents.
+        if vs_comm.process is not None:
+            vs_comm.process.wait()
```

Comment:
Should we provide a reasonable timeout (e.g. 5 seconds) so that if the process hangs, we still dump the logs? Maybe even send a kill?

Reply:
I'd say 20 seconds. Some build bots make every test run very slowly

Reply:
I'm fine with 20 (or really anything below 600, which is the individual lit timeout).

Reply:
The test itself usually handles this with the call to `terminate` in

https://github.com/llvm/llvm-project/blob/b251c29af45d3440374f53bb4c1645e5968593f7/lldb/packages/Python/lldbsuite/test/tools/lldb-dap/dap_server.py#L1273

called from:

https://github.com/llvm/llvm-project/blob/b251c29af45d3440374f53bb4c1645e5968593f7/lldb/packages/Python/lldbsuite/test/tools/lldb-dap/lldbdap_testcase.py#L336

But the `read_packet_thread` thread is never joined or waited on so it tends to write the DAP logs whenever it thinks its done with the connection, not when the process has exited.

Reply:
Okay, so the main thread is going to send a SIGTERM asking the process to terminate gracefully. I think the timeout still makes sense (at least here) in case it doesn't, but if we wanted to send a SIGKILL, I agree we should do it where we currently call terminate.